src/smtml/mappings.ml

```diff
@@ -891,9 +890,16 @@ module Make (M_with_make : M_with_make) : S_with_fresh = struct
         match Stack.top_opt s.ctx with
         | None -> assert false
         | Some ctx ->
-          let ctx, assumptions = encode_exprs ctx assumptions in
+          let ctx, assumptions1 = encode_exprs ctx assumptions in
           s.last_ctx <- Some ctx;
-          M.Solver.check s.solver ~ctx ~assumptions
+
+          let usage_before = Mtime_clock.counter () in
+          let res = M.Solver.check s.solver ~ctx ~assumptions:assumptions1 in
+          let usage_after = Mtime_clock.count usage_before in
+
+          Tmp_log_path.write assumptions (Mtime.Span.to_uint64_ns usage_after);
+
+          res
 
```

Comment:
I would suggest renaming the first `assumptions` to `exprs` or `assumption_exprs` or something, and `assumptions1` to `assumptions`, for readability.